flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/diff/DiffType.java

```diff
@@ -28,7 +28,9 @@ public enum DiffType {
     /** Scalable spec change. */
     SCALE,
     /** Upgradable spec change. */
-    UPGRADE;
+    UPGRADE,
+    /** Full redeploy from new state. */
+    SAVEPOINT_REDEPLOY;
```

Comment:
Why do we need a new state? Isn't that also an `UPGRADE`? I suppose we could also special case inside the upgrade path when the nonce has been set. We also use `UPGRADE` for normal restarts.

Reply:
This allows us to distinguish this more cleanly in a single place instead of having to add some new logic to compare with the lastReconciled spec etc.

This is not really an upgrade in how it was understood before. It doesn't use / preserve state and also does not respect he upgradeMode. Special casing under UPGRADE would probably be more confusing to users and make it harder to test implement.

Reply:
Also this enum is not part of the CRD/status anywhere so it's basically an internal implementation detail to make the logic nice and clean